### DISCUSSION TITLE
Unpredictable behaviour while testing with Jail

### DISCUSSION THREAD

**Discussion by jaads** [figmentapp/figment#11] [Q&A]
I have a couple of tests in which I used `Figment::Jail::expect_with` but I still get errors sometimes due to some sort of race conditions.

I could remove the error by running the test sequentially using the `serial_test` crate and also by using `jail.clear_env();` before starting with the actual test. So my tests look like this now:

```rust
#[test]
#[serial]
fn environment_variables_override_defaults() {
    figment::Jail::expect_with(|jail| {
        jail.clear_env();
        jail.set_env("port", 8888);
        jail.set_env("host", "someotherhost");
        let res = Config::load().unwrap();
        assert_eq!(res.port, 8888);
        assert_eq!(res.host, "someotherhost");
        Ok(())
    });
}
```

I wonder if I'm doing something wrong. Shouldn't the Jail ensure that the test run independently in isolation?

**Comment by fdb** [comment]
I think you're referring to the wrong project...

**Reply by jaads** [reply]
true. sorry!

**Reply by jaads** [reply]
the admin of the repo should be able to delete it. sorry again for the inconvenience!
